spec/02-integration/03-db/03-plugins_spec.lua

```diff
@@ -60,6 +72,23 @@ for _, strategy in helpers.each_strategy() do
                       [[name="key-auth",route={id="]] .. route.id ..
                       [["},consumer=null}']], err_t.message)
         end)
+
+        it("returns an error when inserting mismatched plugins", function()
+          local plugin, _, err_t = db.plugins:insert({ name = "key-auth",
+                                                       protocols = { "http" },
+                                                       route = { id = route.id },
+                                                     })
+          assert.is_nil(plugin)
+          assert.equals(err_t.fields.protocols, "Plugin protocols must match the associated route's protocols")
```

Comment:
suggestions on the error message:
* `Plugin protocols` is redundant given its context